File: tests/expectations/function_ptr.zig
const std = @import("std");

pub const MyCallback =  : ?fn(a: usize, b: usize) anyopaque;

pub const MyOtherCallback =  : ?fn(a: usize,
                                   lot: usize,
                                   of: usize,
                                   args: usize,
                                   and_then_some: usize) anyopaque;

pub extern fn my_function(a: MyCallback, b: MyOtherCallback) callconv(.C) void;

File: tests/expectations/function_ptr.java
import com.sun.jna.*;
import com.sun.jna.ptr.*;

enum BindingsSingleton {
  INSTANCE;
  final Bindings lib = Native.load("", Bindings.class);
}

interface Bindings extends Library {
  Bindings INSTANCE = BindingsSingleton.INSTANCE.lib;

  interface MyCallback extends Callback {
    void invoke(NativeLong a, NativeLong b);
  }

  interface MyOtherCallback extends Callback {
    void invoke(NativeLong a, 
                NativeLong lot, 
                NativeLong of, 
                NativeLong args, 
                NativeLong and_then_some);
  }

  void my_function(MyCallback a, MyOtherCallback b);

}